src/number.cr

```diff
@@ -268,6 +268,35 @@ struct Number
     self == 0
   end
 
+  # Returns a string representation of this number including a suffix
+  # that shows it's type. The suffix is not shown for `Int32` and `Float64`
+  # because they are the default integer and float types respectively, in Crystal.
+  #
+  # ```
+  # 1.dump     # => "1"
+  # 1_i64.dump # => "1_i64"
+  # 1_u8.dump  # => "1_u8"
+  # ```
+  def dump : String
+    String.build do |io|
+      dump(io)
+    end
+  end
+
+  # Appends a string representation of this number including a suffix
+  # that shows it's type. The suffix is not shown for `Int32` and `Float64`
+  # because they are the default integer and float types respectively, in Crystal.
+  #
+  # See `#dump`.
+  def dump(io : IO)
```

Comment:
I would expect to `dump` emit something that I can copy/paste in crystal code and runs.
That is not true for the Big* family that does not have a built-in literal.
What about adding a `dump_prefix` (or change the `#dump` implementation in Big*) so they can be emitted as `BigInt.new("` + `.to_s` + `")` ?

Reply:
My expectations differ in this regard, I'd expect to clearly see the types (see PHP's [`var_dump`](http://php.net/manual/en/function.var-dump.php)), not necessarily be able to copy-paste it - I'd rather use `#inspect` for that.

Reply:
> I'd rather use `#inspect` for that

But you don't always have the correct types then, since there is no type suffix on `#inspect`

Reply:
@bew Oh, right. I guess I don't copy-paste too much `#inspect`-ed output... 😅

Reply:
Well, currently you can copy&paste `inspect` output of numbers. Only when this is merged, you can't.

Reply:
The whole argument about copy pasting is for me pretty weak since how often you actually need to do so? And such behavior would IMO rather fit into `#serialize/deserialize` terms and not `#inspect` or `#dump` - these, as I understand them, are provided for debugging purposes and mainly to be used for visual grepping, not copy-paste.

Reply:
> how often you actually need to do so?
Never for me.

Reply:
> I would expect to dump emit something that I can copy/paste in crystal code and runs.
That is not true for the Big* family that does not have a built-in literal.

And this is why I think we shouldn't provide a `dump` method at all, because it will always be incomplete.